paddle/math/Matrix.cpp

```diff
@@ -2000,6 +2034,23 @@ void CpuMatrix::collectBias(Matrix& a, real scale) {
   }
 }
 
+void CpuMatrix::collectSharedBias(Matrix& a, real scale) {
+  CHECK_EQ(getHeight(), (size_t)1);
+  real* B = getData();
+  real* A = a.getData();
+  size_t numSamples = a.getHeight();
+  size_t channel = getWidth();
+  CHECK_EQ(a.getWidth() % channel, 0UL);
+  size_t dim = a.getWidth() / channel;
+  for (size_t i = 0; i < numSamples; i++) {
+    for (size_t c = 0; c < channel; c++) {
+      for (size_t j = 0; j < dim; j++) {
+        B[c] += scale * A[i * channel * dim + c * dim + j];
```

Comment:
you can compute the channel*dim outside the loop,which can reduce the computation. May be 
```
size_t channel_dim =  channel * dim;
for (size_t i = 0; i < numSamples; i++) {
   size_t index = channel_dim * i;
    for (size_t c = 0; c < channel; c++) {
       size_t index2 = c * dim;
       for (size_t j = 0; j < dim; j++) {
            B[c] += scale * A[index + index2 + j];
```